### DISCUSSION TITLE
Usage of the swapi.tech for a demo in a PowerShell book

### DISCUSSION THREAD

**Discussion by mattcorr** [semperry/swapi#3] [General]
Hello.

I am currently writing a chapter in a collaboration book on PowerShell. I was looking for a live API to use as a sample endpoint to demonstrate the usage of Pester to test PowerShell code, and I came across your site.

Would you allow your https://www.swapi.tech to be used? It would not increase the traffic significantly, as it would require book readers to check out a sample repo and run the code. It is more likely they would just read the book and look at the code rather than run it.

Let me know your thoughts.

Thanks!

Matt

**Comment by semperry** [comment]
Thanks for reaching out, Matt! Sure, I'd be all about it. Keep in mind that
I haven't aliased my domain, so make sure to spell it all the way out when
making the request (https://www). Best practices should be in place by now,
but I didn't expect things to blow up as big as they have and so if you see
changes that could be made just fork and open a pull request. Same goes for
the readers. I'm definitely happy to merge features if people want to fork
and contribute. Finally, do please consider donating if you can, but of
course there is zero obligation or need to do so. I'd love to have a look
at what you put together when it's done!

Let me know if you have any questions in general about the api. Thanks
again.


Ryan Curtis

**Comment by mattcorr** [comment]
Hey again. After some consultation with others working on the book, it was decided to just create our own version of the REST API. It would only be a subset and enough for the demo to work. Plus it will provide me with good experience with Azure functions.

Best of luck with your site and the future.

**Comment by semperry** [comment]
Super cool! Thank you. Good luck to you as well.
